hal/src/sercom/v2/uart/mod.rs

```diff
@@ -0,0 +1,1531 @@
+pub type Clock = pac::MCLK;
+
+impl<P: ValidPads> Config<P> {
+    /// Create a new [`Config`] in the default configuration
+    ///
+    /// This function will enable the corresponding APB clock, reset the
+    /// [`Sercom`] peripheral, and return a [`Config`] in the default
+    /// configuration:
+    ///
+    /// * [`EightBit`] [`CharSize`]
+    /// * No parity
+    /// * One stop bit
+    /// * LSB-first
+    ///
+    /// [`Config`] takes ownership of the [`Sercom`] and [`Pads`].
+    ///
+    /// Users must configure GCLK manually. The `freq` parameter represents the
+    /// GCLK frequency for this [`Sercom`] instance.
+    pub fn new(clk: &Clock, mut sercom: P::Sercom, pads: P, freq: impl Into<Hertz>) -> Self {
+        sercom.enable_apb_clock(clk);
+        Self::default(sercom, pads, freq).msb_first(false)
+    }
+
+    /// Create a new [`Config`] in the default configuration
+    #[inline]
+    fn default(sercom: P::Sercom, pads: P, freq: impl Into<Hertz>) -> Self {
+        let mut registers = Registers::new(sercom);
+        registers.swrst();
+
+        // Enable internal clock mode
+        registers.configure_mode();
+        registers.configure_pads(P::RXPO as u8, P::TXPO as u8);
+        registers.configure_charsize(EightBit::BITS);
+
+        Self {
+            registers,
+            pads,
+            chsize: PhantomData,
+            freq: freq.into(),
+        }
+    }
+}
+
+impl<P, C> Config<P, C>
+where
+    P: ValidPads,
+    C: CharSize,
+{
+    /// Change the [`Config`] [`CharSize`]
+    #[inline]
+    fn change<C2>(self) -> Config<P, C2>
+    where
+        C2: CharSize,
+    {
+        Config {
+            registers: self.registers,
+            pads: self.pads,
+            chsize: PhantomData,
+            freq: self.freq,
+        }
+    }
+
+    /// Trigger the [`Sercom`]'s SWRST and return a [`Config`] in the
+    /// default configuration.
+    #[inline]
+    pub fn reset(self) -> Config<P> {
+        Config::default(self.registers.free(), self.pads, self.freq)
+    }
+
+    /// Consume the [`Config`], reset the peripheral, and return the [`Sercom`]
+    /// and [`Pads`]
+    #[inline]
+    pub fn free(mut self) -> (P::Sercom, P) {
+        self.registers.swrst();
+        (self.registers.free(), self.pads)
+    }
+
+    /// Change the [`CharSize`]
+    #[inline]
+    pub fn char_size<C2: CharSize>(mut self) -> Config<P, C2> {
+        self.registers.configure_charsize(C2::BITS);
+        self.change()
+    }
+
+    /// Change the bit order of transmission (MSB/LSB first)
+    #[inline]
+    pub fn msb_first(mut self, msb_first: bool) -> Self {
+        self.registers.msb_first(msb_first);
+        self
+    }
+
+    /// Change the parity setting
+    #[inline]
+    pub fn parity(mut self, parity: Option<Parity>) -> Self {
+        self.registers.parity(parity);
+        self
+    }
+
+    /// Change the stop bit setting
+    #[inline]
+    pub fn stop_bit(mut self, stop_bit: StopBits) -> Self {
```

Comment:
Super-minor nitpicky observation: it would be more consistent to always use plural `stop_bits()` etc